### ISSUE TITLE
Matplotlib 3.5.1 throws error

### ISSUE THREAD

**Issue by danwos** [useblocks/sphinxcontrib-needs#517]
With Matplotlib 3.5.1. needpie/bar throws an error:

```
/usr/local/lib/python3.8/dist-packages/matplotlib/axes/_axes.py:3052: RuntimeWarning: invalid value encountered in true_divide
x = x / sx



Extension error (sphinxcontrib.needs.directives.needpie):
Handler <function process_needpie at 0x7f795b361040> for event 'doctree-resolved' threw an exception (exception: cannot convert float NaN to integer)
```

Looks like some Matplotlib internal transformation got deactivated and we should do it now on our own.

Version 3.4.3 is still working.

**Comment by danwos** [comment]
That's a tough one.
I was not able to trigger this bug.

Added a lot of strange needpie configurations to our test cases, but they can all be built (long title, labels, all labels on the same place, empty labels, strange function return values, ...)

**Comment by danwos** [comment]
PR #519 will maybe solve this.
Lets hope.

**Comment by twodrops** [comment]
Fix from @aris-bb solves the issue. Thanks

**Comment by danwos** [comment]
Fixed by #519
